xarray/core/dataarray.py

```diff
@@ -1381,8 +1383,29 @@ def transpose(self, *dims):
         numpy.transpose
         Dataset.transpose
         """
+        if dims:
+            if set(dims) ^ set(self.dims):
+                raise ValueError('arguments to transpose (%s) must be '
+                                 'permuted array dimensions (%s)'
+                                 % (dims, tuple(self.dims)))
+
+        transpose_coords = kwargs.pop('transpose_coords', None)
```

Comment:
Should we raise an error when keyword args other than `transpose_coords` are being passed? `kwargs` is necessary for Python 2 compatibility.

Reply:
if we wait 40 days before mergin this, we wouldn't have to worry about Python 2 compatibility at all :)

But yes, for now:
```
if kwargs:
    raise TypeError('unrecognized keyword arguments')
```